### PR TITLE
[Merged by Bors] - feat: port Data.Polynomial.UnitTrinomial

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Polynomial/UnitTrinomial.lean [hunk 1]
```diff
@@ -131,7 +131,7 @@ variable (p q : ℤ[X])
 
 /-- A unit trinomial is a trinomial with unit coefficients. -/
 def IsUnitTrinomial :=
-  ∃ (k m n : ℕ) (hkm : k < m) (hmn : m < n) (u v w : Units ℤ), p = trinomial k m n u v w
+  ∃ (k m n : ℕ) (_hkm : k < m) (_hmn : m < n) (u v w : Units ℤ), p = trinomial k m n (u : ℤ) v w
```
Comment (semorrison) [inline Mathlib/Data/Polynomial/UnitTrinomial.lean:134]:
```suggestion
  ∃ (k m n : ℕ) (_ : k < m) (_ : m < n) (u v w : Units ℤ), p = trinomial k m n (u : ℤ) v w
```
I prefer not to use `_h` unless actually necessary, but I really don't mind either way.

File: Mathlib/Data/Polynomial/UnitTrinomial.lean [hunk 2]
```diff
@@ -181,156 +181,165 @@ theorem isUnitTrinomial_iff :
...
+  rw [Finsupp.filter_add, Finsupp.filter_add, Finsupp.filter_add, Finsupp.filter_add,
+    Finsupp.filter_add, Finsupp.filter_add, Finsupp.filter_add, Finsupp.filter_add]
   rw [Finsupp.filter_single_of_neg, Finsupp.filter_single_of_neg, Finsupp.filter_single_of_neg,
     Finsupp.filter_single_of_neg, Finsupp.filter_single_of_neg, Finsupp.filter_single_of_pos,
     Finsupp.filter_single_of_neg, Finsupp.filter_single_of_pos, Finsupp.filter_single_of_neg]
-  · simp only [add_zero, zero_add, of_finsupp_add, of_finsupp_single]
-    rw [C_mul_monomial, C_mul_monomial, mul_comm ↑v ↑w, add_comm (n - m + k) n]
-  · exact fun h => h.2.Ne rfl
+  · simp only [add_zero, zero_add, ofFinsupp_add, ofFinsupp_single]
+    -- Porting note: added next two lines (less powerful `simp`).
```
Comment (semorrison) [inline Mathlib/Data/Polynomial/UnitTrinomial.lean:239]: I know it's late in the day, but it's super helpful if we include the original proof step in the tactic, so to diagnose regressions we don't have to open mathlib3 and find the corresponding point.

**Comment by semorrison** [comment]
bors d+

File: Mathlib/Data/Polynomial/UnitTrinomial.lean (hunk 2 above)
Comment (Ruben-VandeVelde) [reply Mathlib/Data/Polynomial/UnitTrinomial.lean:239]: The original code is right there, I just added new code, so if simp becomes stronger again, the rewrites will fail.

**Pull request closed without merging**